bin/shopify

```diff
@@ -14,6 +14,15 @@ module Kernel
     raise if name == "#{RUBY_VERSION[/\d+\.\d+/]}/psych.so"
     # Special case for ffi, which rescues this itself
     raise if name == "#{RUBY_VERSION.split(".")[0, 2].join(".")}/ffi_c"
+    # Special case for nokogiri, which might install the wrong architecture
+    if name == "nokogiri/nokogiri"
+      STDERR.puts "[Note] Nokogiri is failing to load, which most likely means " \
+        "it is installed with the wrong architecture for your system. This link " \
+        "has instructions for how to install the correct version for your system: " \
+        "https://nokogiri.org/tutorials/installing_nokogiri.html"
+      STDERR.puts e.full_message
```

Comment:
I'm using an uber-simple method of displaying this, as it's in `require` so I don't want to make assumptions around what has or hasn't been loaded.

Reply:
Another option you might choose is to use `Kernel#warn`: https://ruby-doc.org/core-3.1.0/Kernel.html#method-i-warn

Reply:
You may also want to consider line-wrapping using something like:

```ruby
  warn(<<~EOF)
    [Note] Nokogiri is failing to load.

    This may mean that a gem is not installed correctly, or that the incorrect architecture
    for your system is installed. This link has instructions for how to install the correct
    version for your system:

        https://nokogiri.org/tutorials/installing_nokogiri.html

  EOF
```

Reply:
Ah, the old heredoc. Excellent point!

Reply:
@MeredithCastile we should probably get your thoughts on what's the right error message...

Reply:
@amcaplan @gonzaloriestra I see that you merged this, Gonzalo. Which is the right call, given my speed of response!

For a subsequent release, you might consider this messaging and formatting.

<img width="516" alt="image" src="https://user-images.githubusercontent.com/51347120/154323992-3150a332-17fd-4e95-bafe-84131a9545c7.png">

> The Nokogiri gem is failing to load, due to an installation or architecture issue.
> 
> To fix this, reinstall Nokogiri.
> 
> Installation guide: https://nokogiri.org/tutorials/installing_nokogiri.html

Reply:
I'll recommend not line-wrapping as the individual lines are fairly short, and copying a link with a line break in the middle is likely to lead to user frustration.

Reply:
PR with the new message: https://github.com/Shopify/shopify-cli/pull/2070

Reply:
Thanks, @amcaplan and @gonzaloriestra! Sounds good to me.